lib/src/executor_impl.dart

```diff
@@ -119,8 +122,8 @@ class _Executor implements Executor {
   }
 
   @override
-  Future join({bool withWaiting = false}) {
-    final futures = <Future>[];
+  Future<void> join({bool withWaiting = false}) {
```

Comment:
The method ends with `Future.wait` which returns with a list:
https://api.dart.dev/stable/2.19.1/dart-async/Future/wait.html

Maybe the `package:postgres` issue comes from the lack of type annotation here, and the return type should be `Future<List<Object?>>` instead. It seems to be better to expose the return type here, than to make type casts in the schedule, which 3-6 months from now would seem rather mysterious.

Reply:
I leave that to your decision. I typed it as void because the output of join is not used.
The function could also be marked as async and instead of returning the future we can do `await Future.wait(...)` so that the output is void